crates/core/tedge_agent/src/restart_manager/actor.rs

```diff
@@ -155,12 +155,18 @@ impl RestartManagerActor {
             return response.with_status(OperationStatus::Failed);
         }
 
+        if let Err(err) = create_tmp_restart_file(&self.config.tmp_dir).await {
+            error!(
+                "Fail to create a witness file in {} due to: {}",
+                self.config.tmp_dir, err
+            );
+            return response.with_status(OperationStatus::Failed);
+        }
+
```

Comment:
The call to `create_tmp_restart_file` has been move here to ensure that all the reboot witness files are created before is sent the message  that notifies that the command is executing.